### PR TITLE
Feature/nested identifiers

### PR DESCRIPTION
**Pull request by anusharanganathan**
Added nested identifiers for existing data models. This field is also needed for RDSS dataset.

### CODE REVIEW

File: willow/app/inputs/object_identifier_input.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      required = true
+    end
+
+    # --- obj_id_scheme
+    field = :obj_id_scheme
+    field_name = name_for(attribute_name, index, field)
+    field_id = id_for(attribute_name, index, field)
+    field_value = object_identifier_statement.send(field).first
+
+    out << "<div class='row'>"
```
Comment (martyn-w) [inline willow/app/inputs/object_identifier_input.rb]: Would it be possible to move this html generation code to a rails `.erb` or `.haml` template?, and then call with render('template_name') etc ?

If not easy, its no biggy.

File: willow/app/renderers/nested_identifier_attribute_renderer.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+class NestedIdentifierAttributeRenderer < Hyrax::Renderers::AttributeRenderer
+  private
+  def attribute_value_to_html(value)
```
Comment (martyn-w) [inline willow/app/renderers/nested_identifier_attribute_renderer.rb]: could this be moved to an `.erb` template ?

**Comment by martyn-w** [comment]
Hi @anusharanganathan, the code looks good. I've made a small query about generating the html inline in the code, rather than in an `.erb` file, but its not a big issue and we can live with the code as is.

I need to run and test it, which I can do first thing tomorrow (Wednesday), unless someone else can test it today?

File: willow/app/inputs/object_identifier_input.rb (hunk 1 above)
Comment (anusharanganathan) [reply willow/app/inputs/object_identifier_input.rb]: I know right. I too found this odd. I've extended the Hyrax code base, but stuck to their style. Hyrax has defined all input fields in app/inputs in .rb files. I would say let's leave it for now.

File: willow/app/renderers/nested_identifier_attribute_renderer.rb (hunk 2 above)
Comment (anusharanganathan) [reply willow/app/renderers/nested_identifier_attribute_renderer.rb]: Similarly with the renderers, Hyrax use a .rb file and place them all in app/renderers. We can leave it this way for now, as we are doing it the Hyrax way.

File: willow/app/inputs/object_identifier_input.rb (hunk 1 above)
Comment (martyn-w) [reply willow/app/inputs/object_identifier_input.rb]: 👍  ok sure

**Comment by martyn-w** [comment]
👍  tested and looks good

**Pull request merged**